### PR TITLE
Handle invalid commands.

### LINKED ISSUE #3: Handling command inputs while not in battle
Town commands such as checking your inventory, crafting, and etc. should not be available during battles.

### To-Do
- Restrict the player's ability to access town commands while in a battle
- Send error message letting the player know so

### PR DESCRIPTION
**Pull request by foxfriends**
Closes #3.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -34,44 +35,66 @@ let player = {
...
+    return response;
+  },
+
+  check: () => check(enemy),
+};
+
+const ALL_COMMANDS = new Set([
+  ...Object.keys(TOWN_COMMANDS),
+  ...Object.keys(FIGHT_COMMANDS),
+]);
```
Comment (foxfriends) [inline src/index.js:64]: Today you learn about [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)! Set and `Map` are pretty similar (both are built in classes), except `Set` does not associate any value to the key, it just says "yeah I got that key" or "nah don't got that key" when you call `set.has(somekey)`.

`Set` and array are also very similar, you could totally have done this with an array and `array.includes()`, but Set has better performance at large sizes, due to the searching problem we discussed before. `array.includes()` has no strategy for finding things except for checking elements one by one (the array is not sorted it cannot do the binary search I told you about before), but `Set` is able to do that with direct access (more like making an object `objset = { x: true, y: true }` and then `set.has('x')` is like doing `objset['x']`) -- no searching required.

File: src/index.js [hunk 2]
```diff
@@ -34,44 +35,66 @@ let player = {
...
+  attack: () => {
+    const response = attack(player, enemy, inventory);
+    if (enemy.hp <= 0) {
+      enemy = null;
+    }
+    return response;
+  },
+
+  check: () => check(enemy),
+};
```
Comment (foxfriends) [inline src/index.js:59]: You did some backend stuff with Express right? We're going to slowly move towards building something similar to Express (yes *building* Express) for message handlers. It'll be fun I swear. Express is not that complicated on the inside.

More accurately, it'll be more like [Koa](https://koajs.com/), which is sort of the successor to Express, if you are interested in looking at that. Koa is very similar but easier to use than Express, so if you were going to do another web server I'd recommend it.

Comment (foxfriends) [reply src/index.js:59]: Related to this, I've assigned you to the new issue #14 which is step one of doing this ^

File: src/index.js (hunk 1 above)
Comment (TerrenceJChan) [reply src/index.js:64]: Ohh I like the search strategy that we talked about! I'll do some reading on `Set` and see if I can get accustomed to it. More than anything, I still have to become comfortable with the three dots. From what you describe, I can definitely see the benefits of using `Set` though!

File: src/index.js (hunk 2 above)
Comment (TerrenceJChan) [reply src/index.js:59]: Yes, we worked a lot with Express. Koa is new to me and I'll have a look into it! Looking into #14 now.

**Review by TerrenceJChan — APPROVED** [review]

**Pull request merged**